File: src/org/cerion/symcalc/expression/function/core/Hold.java
package org.cerion.symcalc.expression.function.core;

import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.function.Function;
import org.cerion.symcalc.expression.function.FunctionExpr;

public class Hold extends FunctionExpr {

    public Hold(Expr... e) {
        super(Function.HOLD,e);
    }

    @Override
    protected Expr evaluate() {
        return this;
    }

    @Override
    protected int getProperties() {
        return Properties.HOLD.getValue();
    }
}

File: src/expression/function/core/Hold.kt
package org.cerion.symcalc.expression.function.core

import org.cerion.symcalc.expression.Expr
import org.cerion.symcalc.expression.function.FunctionExpr

class Hold(vararg e: Expr) : FunctionExpr(*e) {

    override val properties: Int
        get() = Properties.HOLD.value

    override fun evaluate(): Expr {
        return this
    }

    override fun validate() {
    }
}
